src/inc/ex.h

```diff
@@ -1306,13 +1285,11 @@ Exception *ExThrowWithInnerHelper(Exception *inner);
     ANNOTATION_HANDLER_END;                              \
     if (IsCurrentExceptionSO())                          \
         __state.SetCaughtSO();                           \
-    VM_NO_SO_INFRASTRUCTURE_CODE(_ASSERTE(!__state.DidCatchSO());) \
+    _ASSERTE(!__state.DidCatchSO());                     \
```

Comment:
Forgot to add this guard back, will fix.

Reply:
I think all this `VM_NO_SO_INFRASTRUCTURE_CODE` or `DidCatchSO` can be deleted as well. It is all related to the stack probing and stack overflow handling that does not exist in .NET Core.

Reply:
(can be done in follow up change)